### ISSUE TITLE
Unable to place RoboDozer in 3 out of 4 visible map corners

### ISSUE THREAD

**Issue by belgianguy** [OutpostUniverse/OPHD#788]
Test situation:
After starting a new game and placing the lander, advanced a turn and I'm trying to place a RoboDozer at each of the four visible corners of the visible map per turn:
![cant_bulldoze](https://user-images.githubusercontent.com/3091512/108596618-7c1cb480-7386-11eb-8f2a-1d808c5fe1b4.png)

In the current setup, the only corner I could place it was at the top.

Perhaps this is a design decision or something the original also did or an unwritten law of isometric design, if so, the issue can be summarily closed.

The other three corners did not allow me to. It seems not possible to place a RoboDozer at the SE and SW edges (also marked in red). I also resized my window to get both an odd and even edge length, in both cases the above situation was true.

I guess this has to do with how the game decides which tile is currently visible. Even though it is visible to me as a human, the game code tells me it is in fact not visible to the game. The top corner that did work maps to the origin of the Rectangle, all other 3 corners are literal "edge" cases, where one or both coordinates are just outside of the calculated rectangle, even though they are (humanly) visible tiles. 

```
bool TileMap::isVisibleTile(NAS2D::Point<int> position, int z) const
{
	if (!NAS2D::Rectangle{mMapViewLocation.x, mMapViewLocation.y, mEdgeLength - 1, mEdgeLength - 1}.contains(position))
	{
		return false;
	}

	if (z != mCurrentDepth)
	{
		return false;
	}

	return true;
}
```

So would that mean the mEdgeLength should not be decreased by 1 in both places when checking a position with contains ?

**Comment by DanRStevens** [comment]
Ahh, thank you for reporting this. That does indeed look like a bug.

I think I'd noticed this before, though was kind of busy with other things and forgot about it.

I think I had played around with adjusting the visibility code to remove the `- 1`, though I may have run into issues somewhere else when I tried that. I can't quite remember now. Though it's possible perhaps I was adjusting the wrong function. I was much less familiar with the code back then.
